docs/docsite/rst/userguide/inventories.rst

```diff
@@ -1084,6 +1085,57 @@ Red Hat Ansible Automation Platform
 
 4. Use the **Source Variables** field to override variables used by the ``controller`` inventory plugin. Enter variables using either JSON or YAML syntax. Use the radio button to toggle between the two. 
 
+
+.. _ug_source_terraform:
+
+Terraform State
+~~~~~~~~~~~~~~~~
+
+.. index::
+   pair: inventories; Terraform
+   pair: inventory source; Terraform state
+
+
+This inventory source uses the `terraform_state <https://github.com/ansible-collections/cloud.terraform/blob/main/plugins/inventory/terraform_state.py>`_ inventory plugin from the `cloud.terraform <https://github.com/ansible-collections/cloud.terraform>`_ collection. The plugin will parse a terraform state file and add hosts for AWS EC2, GCE, and Azure instances.
+
+1. To configure this type of sourced inventory, select **Terraform State** from the Source field.
+
+2. The Create new source window expands with the required **Credential** field. Choose from an existing Terraform backend Credential. For more information, refer to :ref:`ug_credentials`.
+
+3. You can optionally specify the verbosity, host filter, enabled variable/value, and update options as described in the main procedure for :ref:`adding a source <ug_add_inv_common_fields>`. For Terraform, enable **Overwrite** and **Update on launch** options.  
+
+4. Use the **Source Variables** field to override variables used by the ``controller`` inventory plugin. Enter variables using either JSON or YAML syntax. Use the radio button to toggle between the two. For more information on these variables, see the `terraform_state <https://github.com/ansible-collections/cloud.terraform/blob/main/plugins/inventory/terraform_state.py>`_ file for detail.
+
+  The ``backend_type`` variable is required by the Terraform state inventory plugin. This should match the remote backend configured in the Terraform backend credential, here is an example for an s3 backend:
+
+  ::
+
+    ---
+    backend_type: s3
+
+5. Enter an |ee| in the **Execution Environment** field that contains a Terraform binary. This is required for the inventory plugin to run the Terraform commands that read inventory data from the Terraform state file.
+
+  .. image:: ../common/images/inventories-create-source-terraform-example.png
+      :width: 1400px
+
+6. To add hosts for AWS EC2, GCE, and Azure instances:
```

Comment:
I totally get why this step was added but I think it's a bit confusing, as it doesn't cover all of what needs to happen in that job template to deploy those resources via Terraform. I wonder if it might be better to just say here that the Terraform state file in the backend must contain state for resources already deployed to EC2, GCE, or Azure. And then we can point to other documentation on how to do that. @gravesm any thoughts?

Reply:
My inclination would be to say something like "Refer to the AWS, GCE, or Azure Terraform providers to provision instances." If we want to link to the documentation for those providers we can, but I don't think we should maintain docs on how to use them.

Reply:
^^^Agree!